asf_heat_pump_suitability/research/exploratory/decision_tree_categorisation/applying_decision_tree.py

```diff
@@ -0,0 +1,1332 @@
+# Number of unique land parcels in Plymouth vs. number of buildings
+# This shows that lots of land parcels are aggregated into one single building geometry/ footprint
+plymouth_gdf["NATIONALCADASTRALREFERENCE"].nunique(), plymouth_gdf[
+    "building_geometry"
+].nunique()
+
+# %%
+# Identifying pairs of different 1st most suitable solutions per land parcel
+solutions_per_land_parcel = plymouth_gdf.groupby("NATIONALCADASTRALREFERENCE")[
+    "1st_most_suitable_solution"
+].apply(set)
+solutions_per_land_parcel = pd.DataFrame(solutions_per_land_parcel)
+solutions_per_land_parcel["n_solutions"] = solutions_per_land_parcel[
+    "1st_most_suitable_solution"
+].apply(len)
+solutions_per_land_parcel = solutions_per_land_parcel[
+    solutions_per_land_parcel["n_solutions"] > 1
+]
+solutions_per_land_parcel.reset_index(inplace=True)
+solutions_per_land_parcel["1st_most_suitable_solution_str"] = solutions_per_land_parcel[
+    "1st_most_suitable_solution"
+].apply(lambda x: ", ".join(x))
+solutions_per_land_parcel.groupby("1st_most_suitable_solution_str")[
+    ["NATIONALCADASTRALREFERENCE"]
+].nunique()
+
+
+# %%
+# Identifying pairs of different 1st most suitable solutions per building footprint
+solutions_per_footprint = plymouth_gdf.groupby("building_geometry")[
+    "1st_most_suitable_solution"
+].apply(set)
+solutions_per_footprint = pd.DataFrame(solutions_per_footprint)
+solutions_per_footprint["n_solutions"] = solutions_per_footprint[
+    "1st_most_suitable_solution"
+].apply(len)
+solutions_per_footprint = solutions_per_footprint[
+    solutions_per_footprint["n_solutions"] > 1
+]
+solutions_per_footprint.reset_index(inplace=True)
+solutions_per_footprint["1st_most_suitable_solution_str"] = solutions_per_footprint[
+    "1st_most_suitable_solution"
+].apply(lambda x: ", ".join(x))
+solutions_per_footprint.groupby("1st_most_suitable_solution_str")[
+    ["building_geometry"]
+].nunique()
+
+
+# %%
+# check if each land parcel appears only within one building footprint
+land_parcel_to_footprint = plymouth_gdf.groupby("NATIONALCADASTRALREFERENCE")[
+    "building_geometry"
+].apply(set)
+land_parcel_to_footprint = pd.DataFrame(land_parcel_to_footprint)
+land_parcel_to_footprint["n_footprints"] = land_parcel_to_footprint[
+    "building_geometry"
+].apply(len)
+
+# %%
+# It doesn't seem to be the case
+land_parcel_to_footprint["n_footprints"].value_counts()
+
+# %%
+land_parcel_to_footprint["n_footprints"].value_counts(normalize=True) * 100
+
+# %% [markdown]
+# In most cases:
+# - the same land parcel should be in one building footprint only (hopefully?)
+# - properties in the same land parcel should have the same most suitable solution
+# - properties in the same building footprint should have the same most suitable solution
+
+# %%
+# This is currently based on Plymouth only (both pairs found in land parcels and building footprints)
+# but needs to be generalised to all possible cases
+
+
+def assign_unique_sol(solution_set: set) -> str:
```

Comment:
Without listing all possible combinations as I've done for now, it could be a simple as the following:
If DHN in the list of techs: DHN
Elif networked GSHP in the list of techs: networked GSHP
Elif communal solution in the list: communal solution